### DISCUSSION TITLE
[Questions] How to connect strimzi kafka as input/outputs at Azure stream Analytics?

### DISCUSSION THREAD

**Discussion by Mehmoodhassan** [strimzi/strimzi-kafka-operator#10363] [Q&A] [answered]
Hi, 
I have deployed strimzi kafka cluster in my aks cluster and using the default certificates creation mechanism.
Now I want to connect my kakfa as output at Azure stream analytics which requiring parameters as per attched screenshot
![Screenshot from 2024-07-19 18-11-49](https://github.com/user-attachments/assets/7bd9cda3-8588-4ba6-a9a1-8942fe253653)
I have tried a lot solution passing different combination of certificates but unfortunately no luck. I just want to know which set of certificates I need to pass under Truststrore certificates. Below are the available creds
kafkatopic: user-entities-int
kafkauser: qa-test-user-int
kafkauser-secret containing:
  - ca.crt
  - user.crt
  - user.key
  - user.p12
  - user.password

kakfa cluster certificate in this namespace is `aks-dev-cluster-ca-cert` containing below values
  - ca.crt
  - ca.p12
  - ca.password

I will be very thankful to you. Many thanks in advance :)

**Comment by Mehmoodhassan** [comment]
I am eagerly waiting for your response @scholzj

**Reply by scholzj** [reply, marked as answer]
The certificate your Kafka client needs to trust is the one from `aks-dev-cluster-ca-cert`. Depending on what format does the client accept you can use the CRT or the PKCS12 file or convert it to something else. The user certificate depends on your configuration. But if you use TLS client authentication, the `user.crt` and `user.key` are the user certificates.

**Reply by Mehmoodhassan** [reply]
Actually I have created kafka-user and kafka automatically created secret for (containing `ca.crt`, `user,crt` , `user.key` , `user.p12`  and `user.password`) and now I want to configure Azure stream analytics output using that user via `SASL_SSL` protocol but I don't know which certificate I need to pass in Trustore certificates. Moreover, my kafka broker is publically accessible via ingress.

**Reply by scholzj** [reply]
I guess you might need to share your Kafka custom resource / configuration. But in general, if you wanna use SASL_SSL, you do not need a TLS user. You need an SCRAM-SHA-512 user or something to have a username and password.
